nixos/modules/services/desktops/gnome/xkb-settings.nix

```diff
@@ -0,0 +1,91 @@
+# GNOME Settings Daemon
+
+{ config, lib, ... }:
+
+let
+  xcfg = config.services.xserver;
+
+  splitToParts = str: builtins.filter builtins.isString (builtins.split ",\s*" str);
+
+  layouts = splitToParts xcfg.xkbLayout;
+  variantsUnpadded = splitToParts xcfg.xkbVariant;
+  variants =
+    let
+      padding = lib.max 0 (builtins.length layouts - builtins.length variantsUnpadded);
+    in
+    variantsUnpadded ++ lib.replicate padding "";
+in
+
+{
+
+  meta = {
+    maintainers = lib.teams.gnome.members;
+  };
+
+  options = {
+    services.gnome.useXkbConfig = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        If enabled, configure the keyboard layout in GNOME Settings Daemon to match [xserver keyboard settings](#opt-services.xserver.xkbLayout).
+      '';
+    };
+  };
+
+  config = lib.mkIf config.services.gnome.useXkbConfig {
+    assertions = [
+      {
+        /*
+        X appears to ignore the variant option when it is not the same type (string/list) as layout, compare:
+
+            xkbvalidate pc104 us,cz,pl qwerty ""
+            xkbvalidate pc104 us qwerty, ""
+
+        and
+
+            xkbvalidate pc104 us,cz,pl qwerty, ""
+        */
+        assertion = (builtins.length layouts == 1) != (builtins.length variantsUnpadded == 1);
+        message = "services.xserver.xkbLayout and services.xserver.xkbVariant should have the same number of comma-separated items.";
+      }
+    ];
+
+    programs = {
+      dconf = {
+        profiles = {
+          user = {
+            databases = [
+              {
```

Comment:
However, since databases are anonymous, it is not possible to easily add other X-incompatible items – for example, I would like to have:

```nix
(lib.gvariant.mkTuple [
  "ibus"
  "mozc-jp"
])
```

We could add `gnome.keyboard.layouts` option that would also be merged into this but that is not really suitable for multi-user systems. So maybe home-manager is really the proper place for this.